src/FileMutex.php

```diff
@@ -26,39 +28,53 @@ public function __construct(private readonly string $fileName, ?Filesystem $file
         $this->directory = \dirname($this->fileName);
     }
 
+    /**
+     * @throws SyncException
+     */
     public function acquire(?Cancellation $cancellation = null): Lock
     {
         if (!$this->filesystem->isDirectory($this->directory)) {
             throw new SyncException(\sprintf('Directory of "%s" does not exist or is not a directory', $this->fileName));
         }
 
-        // Try to create the lock file. If the file already exists, someone else
-        // has the lock, so set an asynchronous timer and try again.
+        // Try to create and lock the file. If flock fails, someone else already has the lock,
+        // so set an asynchronous timer and try again.
         for ($attempt = 0; true; ++$attempt) {
-            try {
-                $file = $this->filesystem->openFile($this->fileName, 'x');
-
-                // Return a lock object that can be used to release the lock on the mutex.
-                $lock = new Lock($this->release(...));
-
-                $file->close();
+            \set_error_handler(self::$errorHandler ??= static fn () => true);
 
-                return $lock;
-            } catch (FilesystemException) {
-                delay(\min(self::DELAY_LIMIT, self::LATENCY_TIMEOUT * (2 ** $attempt)), cancellation: $cancellation);
+            try {
+                $handle = \fopen($this->fileName, 'c');
+                if ($handle && \flock($handle, \LOCK_EX | \LOCK_NB)) {
```

Comment:
`fopen` is a blocking function, but assuming a local file is being opened this should be a reasonable trade-off, and the only way `flock` can be used in PHP to my knowledge.